docs/aws/audit/redshiftmonitoring/rules/not_using_default_port_remediation.mdx

```diff
@@ -1,6 +1,230 @@
+   - On the cluster details page, click on the "Properties" tab.
+   - Under the "Cluster" section, click the "Edit" button.
+
+4. **Change the Port:**
+   - In the "Cluster configuration" section, find the "Port" field.
+   - Change the port number from the default (5439) to a custom port number of your choice.
+   - Click "Save changes" to apply the new port configuration.
+
+By following these steps, you can ensure that your Redshift cluster does not use the default port, thereby enhancing security by reducing the risk of targeted attacks on the default port.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Redshift clusters from using the default port (5439) using AWS CLI, you can follow these steps:
+
+1. **Describe the Redshift Clusters**:
+   First, identify the Redshift clusters and check their current port settings.
+   ```sh
+   aws redshift describe-clusters --query 'Clusters[*].{ClusterIdentifier:ClusterIdentifier, Port:Port}'
+   ```
+
+2. **Modify the Redshift Cluster Port**:
+   If a cluster is using the default port (5439), modify it to use a different port. For example, to change the port to 5432:
+   ```sh
+   aws redshift modify-cluster --cluster-identifier <your-cluster-identifier> --port 5432
+   ```
+
+3. **Verify the Changes**:
+   After modifying the port, verify that the changes have been applied correctly.
+   ```sh
+   aws redshift describe-clusters --cluster-identifier <your-cluster-identifier> --query 'Clusters[*].{ClusterIdentifier:ClusterIdentifier, Port:Port}'
+   ```
+
+4. **Update Security Groups**:
+   Ensure that the security groups associated with the Redshift cluster allow traffic on the new port. Modify the security group rules if necessary.
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-id <your-security-group-id> --protocol tcp --port 5432 --cidr <your-cidr-block>
+   ```
+
+By following these steps, you can prevent Redshift clusters from using the default port using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Redshift clusters from using the default port (5439) using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+3. **Check Existing Redshift Clusters**:
+   Retrieve the list of existing Redshift clusters and check their port configurations.
+
+4. **Modify Redshift Cluster Port**:
+   If any cluster is using the default port (5439), modify the cluster to use a different port.
+
+Here is a Python script that demonstrates these steps:
+
+```python
+import boto3
+
+# Initialize a Boto3 session
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>